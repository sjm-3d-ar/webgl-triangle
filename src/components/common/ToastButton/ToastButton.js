import React from "react";
import Button from "@material-ui/core/Button";
import { useSnackbar } from "notistack";

const ToastButton = () => {
  const { enqueueSnackbar } = useSnackbar();

  const showToast = () => {
    enqueueSnackbar("Yay, toast!", { variant: "success" });
  };

  return (
    <Button variant="contained" color="primary" onClick={showToast}>
      Show Toast
    </Button>
  );
};

export default ToastButton;
