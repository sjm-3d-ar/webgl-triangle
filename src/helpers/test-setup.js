/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";

import { theme, SnackbarProviderWrapper } from "_helpers";

//
// Set up a custom render, to include ThemeProvider
// example: https://testing-library.com/docs/react-testing-library/setup#custom-render
//

const ProviderWrapper = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <SnackbarProviderWrapper>{children}</SnackbarProviderWrapper>
    </ThemeProvider>
  );
};

const renderWithTheme = (ui, options) =>
  render(ui, { wrapper: ProviderWrapper, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { renderWithTheme as render };
