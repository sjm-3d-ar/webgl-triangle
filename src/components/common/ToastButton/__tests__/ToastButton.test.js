import React from "react";
import { render } from "_helpers/test-setup";

import ToastButton from "../index";

describe("<ToastButton />", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<ToastButton />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should show the ToastButton", () => {
    const { queryByText } = render(<ToastButton />);

    expect(queryByText("Show Toast")).toBeVisible();
  });
});
