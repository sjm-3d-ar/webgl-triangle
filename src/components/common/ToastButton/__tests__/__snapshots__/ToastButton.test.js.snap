// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToastButton /> should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Show Toast
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
